src/main/scala/sigmastate/serialization/ValueSerializer.scala

```diff
@@ -146,13 +163,33 @@ object ValueSerializer extends SigmaSerializerCompanion[Value[SType]] {
     v
   }
 
+  private val SerializedConstantsStartCode = ConstantCode
```

Comment:
@ergomorphic I'm pretty sure we need a separate code here. It marks the serialized constants section. It works using `ConstantCode` (zero) although I sense it should be fragile.

Reply:
@greenhat no, we don't need this code at all <num constants><constants array><script tree>